eslint-plugin-expensify/prefer-at.js

```diff
@@ -0,0 +1,81 @@
+        fixable: 'code',
+    },
+    create(context) {
+        function isLiteralNumber(node) {
+            return node && node.type === 'Literal' && typeof node.value === 'number';
+        }
+
+        function isMemberExpressionWithLength(node) {
+            return (
+                node &&
+                node.type === 'MemberExpression' &&
+                node.property &&
+                node.property.type === 'Identifier' &&
+                node.property.name === 'length'
+            );
+        }
+
+        function checkNode(node) {
+            if (node.type === 'MemberExpression' && node.property) {
+                // Handle items cases like test[0]
+                if (isLiteralNumber(node.property)) {
+                    context.report({
+                        node,
+                        message,
+                        fix(fixer) {
+                            return fixer.replaceText(node, `${node.object.name}.at(${node.property.value})`);
+                        },
+                    });
+                }
+
+                // Handle items like test[test.length - 1]
+                if (
+                    node.property.type === 'BinaryExpression' &&
+                    node.property.operator === '-' &&
+                    isMemberExpressionWithLength(node.property.left) &&
+                    isLiteralNumber(node.property.right)
+                ) {
+                    context.report({
+                        node,
+                        message,
+                        fix(fixer) {
+                            const objectName = node.object.name;
+                            const offset = node.property.right.value;
+                            return fixer.replaceText(node, `${objectName}.at(-${offset})`);
+                        },
+                    });
+                }
+
+                // Handle items like test[(test.length - 1) / 2]
+                if (
+                    node.property.type === 'BinaryExpression' &&
+                    node.property.operator === '/' &&
+                    node.property.left.type === 'BinaryExpression' &&
+                    node.property.left.operator === '-' &&
+                    isMemberExpressionWithLength(node.property.left.left) &&
+                    isLiteralNumber(node.property.left.right) &&
+                    isLiteralNumber(node.property.right)
+                ) {
+                    context.report({
+                        node,
+                        message,
+                        fix(fixer) {
+                            const objectName = node.object.name;
+                            const leftValue = node.property.left.right.value;
+                            const rightValue = node.property.right.value;
+                            return fixer.replaceText(node, `${objectName}.at((${objectName}.length - ${leftValue}) / ${rightValue})`);
+                        },
+                    });
+                }
```

Comment:
These kind of conditions are very specific and will fail if we test with anything that is different from them.. we should either think in a condition that can capture anything inside `[...]` and convert it to `.at(...)`, **if the variable is an array**.